tests/scripts/thread-cert/Cert_5_1_03_RouterAddressReallocation.py

```diff
@@ -72,113 +95,133 @@ def test(self):
+                                  } <= set(p.mle.tlv.type) and\
+                       p.ipv6.hlim == 255 and\
+                       p.mle.tlv.scan_mask.r == 1 and\
+                       p.mle.tlv.scan_mask.e == 1).\
+                must_next()
+
+        # Step 6: Router_1 MUST attempt to attach to any other partition
+        #         within range by sending a MLE Parent Request.
+        #         The following TLVs MUST be present in the MLE Parent Request:
+        #            - Challenge TLV
+        #            - Mode TLV
+        #            - Scan Mask TLV
+        #            - Version TLV
+
+        pkts.filter_wpan_src64(ROUTER_2).\
+                   filter_mle_cmd(MLE_ADVERTISEMENT).\
+                   must_next().\
+                   must_verify(lambda p:\
+                               p.mle.tlv.leader_data.partition_id !=\
+                               _pkt.mle.tlv.leader_data.partition_id)
+
+        pkts.filter_wpan_src64(ROUTER_1).\
+            filter_LLARMA().\
+            filter_mle_cmd(MLE_PARENT_REQUEST).\
+            filter(lambda p: {
+                              CHALLENGE_TLV,
+                              MODE_TLV,
+                              SCAN_MASK_TLV,
+                              VERSION_TLV
+                              } <= set(p.mle.tlv.type) and\
+                   p.ipv6.hlim == 255)\
+            .must_next()
+
+        # Step 7: Router_1 sends a MLE Child ID Request to Router_2.
+        #         The following TLVs MUST be present in the MLE Child ID Request:
+        #             - Link-layer Frame Counter TLV
+        #             - Mode TLV
+        #             - Response TLV
+        #             - Timeout TLV
+        #             - TLV Request TLV
+        #             - Version TLV
+        #             - MLE Frame Counter TLV (optional)
+        #         The following TLV MUST NOT be present in the MLE Child ID Request:
+        #             - Address Registration TLV
+
+        _pkt = pkts.filter_wpan_src64(ROUTER_1).\
+            filter_wpan_dst64(ROUTER_2).\
+            filter_mle_cmd(MLE_CHILD_ID_REQUEST).\
+            filter(lambda p: {
+                              LINK_LAYER_FRAME_COUNTER_TLV,
+                              MODE_TLV,
+                              RESPONSE_TLV,
+                              TIMEOUT_TLV,
+                              TLV_REQUEST_TLV,
+                              VERSION_TLV
+            } <= set(p.mle.tlv.type)).\
+                   must_next()
+        _pkt.must_not_verify(lambda p: (ADDRESS_REGISTRATION_TLV) in p.mle.tlv.type)
+
+        # Step 8: Router_1 sends an Address Solicit Request.
+        #         Ensure the Address Solicit Request is properly formatted:
+        #         CoAP Request URI
+        #             coap://<leader address>:MM/a/as
+        #         CoAP Payload
+        #             - MAC Extended Address TLV
+        #             - Status TLV
```

Comment:
RLOC16 TLV is optional for a/as, however for this case, it should occur as it want to allocate its original rloc16.

Full content in Test Plan is
> CoAP Payload
> MAC Extended Address TLV
> Status TLV
> RLOC16 TLV (optional)

Here the `(optional)` should be removed.

Reply:
added